target/linux/generic/backport-5.10/811-v6.1-0001-nvmem-core-Fix-memleak-in-nvmem_register.patch

```diff
@@ -22,10 +22,23 @@ Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
 
 --- a/drivers/nvmem/core.c
 +++ b/drivers/nvmem/core.c
-@@ -830,21 +830,18 @@ struct nvmem_device *nvmem_register(cons
+@@ -833,6 +833,12 @@ struct nvmem_device *nvmem_register(cons
  	nvmem->dev.groups = nvmem_dev_groups;
  #endif
  
++	if (nvmem->nkeepout) {
++		rval = nvmem_validate_keepouts(nvmem);
++		if (rval)
++			goto err_device_del;
```

Comment:
```suggestion
+			goto err_put_device;
```
`err_device_del` does not exist here.

I compared drivers/nvmem/core.c between v6.1.15 and 5.10.172 + your patches and got this diff:
````
hauke@hauke-t480:~/linux/linux-next-clean$ git diff HEAD v6.1.15 -- drivers/nvmem/core.c
diff --git a/drivers/nvmem/core.c b/drivers/nvmem/core.c
index 02f4a1ab1b75..34ee9d36ee7b 100644
--- a/drivers/nvmem/core.c
+++ b/drivers/nvmem/core.c
@@ -836,7 +836,7 @@ struct nvmem_device *nvmem_register(const struct nvmem_config *config)
        if (nvmem->nkeepout) {
                rval = nvmem_validate_keepouts(nvmem);
                if (rval)
-                       goto err_device_del;
+                       goto err_put_device;
        }
 
        if (config->compat) {
@@ -1237,16 +1237,21 @@ struct nvmem_cell *of_nvmem_cell_get(struct device_node *np, const char *id)
        if (!cell_np)
                return ERR_PTR(-ENOENT);
 
-       nvmem_np = of_get_next_parent(cell_np);
-       if (!nvmem_np)
+       nvmem_np = of_get_parent(cell_np);
+       if (!nvmem_np) {
+               of_node_put(cell_np);
                return ERR_PTR(-EINVAL);
+       }
 
        nvmem = __nvmem_device_get(nvmem_np, device_match_of_node);
        of_node_put(nvmem_np);
-       if (IS_ERR(nvmem))
+       if (IS_ERR(nvmem)) {
+               of_node_put(cell_np);
                return ERR_CAST(nvmem);
+       }
 
        cell_entry = nvmem_find_cell_entry_by_node(nvmem, cell_np);
+       of_node_put(cell_np);
        if (!cell_entry) {
                __nvmem_device_put(nvmem);
                return ERR_PTR(-ENOENT);
hauke@hauke-t480:~/linux/linux-next-clean$ 
````
If you change `err_put_device` it should be fine.